tests/tt_metal/tt_metal/test_kernels/misc/test_fp16_tile_mul_in_risc.cpp

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
how does this test reuse?  the file name sounds like a multiply test - should it be named for the test that runs it?

Reply:
This was added for an end to end stress test that could run this kernel on a full grid and validate outputs from all buffers across all devices. Additionally, I didn't see kernels in the dispatch test infra using the full grid uniformly with  heavy RTA dispatch (unicasts across the entire core grid ... any issues with Virtual Coords could be exposed here). I was trying to encapsulate that in this test.
File has been renamed to `full_grid_eltwise_device_reuse.cpp`